demisto_sdk/commands/validate/validators/PB_validators/PB101_is_playbook_has_unreachable_condition.py

```diff
@@ -0,0 +1,88 @@
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Playbook
+
+
+class IsAskConditionHasUnreachableConditionValidator(BaseValidator[ContentTypes]):
+    error_code = "PB101"
+    description = "Checks whether an ask conditional has task with unreachable next task condition."
+    rationale = "Checks whether an ask conditional has task with unreachable next task condition."
+    error_message = (
+        "Playbook conditional task with id:{task_id} has task with unreachable "
+        "next task condition '{next_task_branch}'. Please remove this task or add "
+        "this condition to condition task with id:{task_id}."
+    )
+    related_field = "tasks"
+    is_auto_fixable = False
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        """Checks whether a builtin conditional task branches are handled properly
+        Args:
+            - content_items (Iterable[ContentTypes]): The content items to check.
+        Return:
+            - List[ValidationResult]. List of ValidationResults objects.
+        """
+        results: List[ValidationResult] = []
+        for content_item in content_items:
+            invalid_tasks = self.invalid_tasks(content_item)
+
+            for task_id, next_task_branch in invalid_tasks.items():
+                results.append(
+                    ValidationResult(
+                        validator=self,
+                        message=self.error_message.format(
+                            task_id=task_id,
+                            next_task_branch=next_task_branch,
+                        ),
+                        content_object=content_item,
+                    )
+                )
+
+        return results
+
+    @staticmethod
+    def invalid_tasks(playbook: ContentTypes):
+        invalid_tasks = {}
+        tasks = playbook.tasks
+        for task in tasks.values():
+            if task.get("type") == "condition" and task.get("message"):
+                next_tasks: dict = task.get("nexttasks", {})
+
+                # ADD all replyOptions to unhandled_reply_options (UPPER)
+                unhandled_reply_options = set(
+                    map(str.upper, task.get("message", {}).get("replyOptions", []))
+                )
+
+                # Rename the keys in dictionary to upper case
+                next_tasks_upper = {k.upper(): v for k, v in next_tasks.items()}
+
+                mapper_dict = {
+                    "YES": ["YES", "TRUE POSITIVE"],
+                    "TRUE POSITIVE": ["YES", "TRUE POSITIVE"],
+                    "NO": ["NO", "FALSE POSITIVE"],
+                    "FALSE POSITIVE": ["NO", "FALSE POSITIVE"],
+                }
+
+                # Remove all nexttasks from unhandled_reply_options (UPPER)
+                for next_task_branch, next_task_id in next_tasks_upper.items():
```

Comment:
same here with nested loops gather all next_tasks_upper and iterate over it without a nested loop.

Reply:
Seems like there is a dependency down the road which prevents breaking the nested for loop